### PR TITLE
[nanoleaf] Removed dependency on org.apache.commons

### PR DESCRIPTION
**Pull request by cweitkamp**
- Removed dependency on org.apache.commons

Related to #7722

Signed-off-by: Christoph Weitkamp <github@christophweitkamp.de>

### CODE REVIEW

File: bundles/org.openhab.binding.nanoleaf/src/main/java/org/openhab/binding/nanoleaf/internal/handler/NanoleafControllerHandler.java [hunk 1]
```diff
@@ -270,11 +270,13 @@ public boolean unregisterControllerListener(NanoleafControllerListener controlle
 
     public NanoleafControllerConfig getControllerConfig() {
         NanoleafControllerConfig config = new NanoleafControllerConfig();
-        config.address = this.getAddress();
+        String localAddress = this.getAddress();
+        config.address = localAddress == null ? "" : localAddress;
```
Comment (J-N-K) [inline bundles/org.openhab.binding.nanoleaf/src/main/java/org/openhab/binding/nanoleaf/internal/handler/NanoleafControllerHandler.java:274]:
```suggestion
        config.address = Objects.requireNonNullElse(getAddress(), "");
```

File: bundles/org.openhab.binding.nanoleaf/src/main/java/org/openhab/binding/nanoleaf/internal/handler/NanoleafControllerHandler.java [hunk 2]
```diff
@@ -270,11 +270,13 @@ public boolean unregisterControllerListener(NanoleafControllerListener controlle
...
         config.port = this.getPort();
         config.refreshInterval = this.getRefreshIntervall();
         config.authToken = this.getAuthToken();
-        config.deviceType = this.getDeviceType();
+        String localDeviceType = this.getDeviceType();
+        config.deviceType = localDeviceType == null ? "" : localDeviceType;
```
Comment (J-N-K) [inline bundles/org.openhab.binding.nanoleaf/src/main/java/org/openhab/binding/nanoleaf/internal/handler/NanoleafControllerHandler.java:279]: see above

File: bundles/org.openhab.binding.nanoleaf/src/main/java/org/openhab/binding/nanoleaf/internal/handler/NanoleafControllerHandler.java [hunk 3]
```diff
@@ -824,8 +830,8 @@ private void sendRhythmCommand(Command command) throws NanoleafException {
         OpenAPIUtils.sendOpenAPIRequest(setNewRhythmRequest);
     }
 
-    private String getAddress() {
-        return StringUtils.defaultString(this.address);
+    private @Nullable String getAddress() {
+        return address;
```
Comment (J-N-K) [inline bundles/org.openhab.binding.nanoleaf/src/main/java/org/openhab/binding/nanoleaf/internal/handler/NanoleafControllerHandler.java:834]: Is this method still needed?

**Review by J-N-K — CHANGES_REQUESTED** [review]

File: bundles/org.openhab.binding.nanoleaf/src/main/java/org/openhab/binding/nanoleaf/internal/handler/NanoleafControllerHandler.java (hunk 3 above)
Comment (cweitkamp) [reply bundles/org.openhab.binding.nanoleaf/src/main/java/org/openhab/binding/nanoleaf/internal/handler/NanoleafControllerHandler.java:831]: No, none of these private getters and setters is needed. But I do not want to refactor the whole binding. I just want to get rid of the dependency.

**Review by J-N-K — APPROVED** [review]

**Review by cpmeister — APPROVED** [review]
LGTM

**Pull request merged**